src/posts/react-native-monorepo.md

```diff
@@ -0,0 +1,183 @@
+title: "Migrating to a Mobile Monorepo for React Native"
+published_on: May 29, 2017
+author: Jared Forsyth
+team: Mobile
+...
+
+Over the past few months, we've been adding some [React
+Native](https://github.com/facebook/react-native) to our existing
+[iOS](https://itunes.apple.com/us/app/khan-academy-you-can-learn-anything/id469863705?mt=8)
+and
+[Android](https://play.google.com/store/apps/details?id=org.khanacademy.android&hl=en)
+apps. We started out by just creating a `react-native` repository and having
+it be a submodule of our respective `ios` and `android` git repositories, but
+we quickly found that there was a fair amount of friction in keeping things
+coordinated between the three. We've now moved all of our mobile-related
+repositories (including a `mobile-scripts` and a `shared-webview` repository)
+into a single `mobile` [monorepo](https://danluu.com/monorepo/).
+
+## Why did we do it?
+
+When making changes to the js↔native FFI, we need to make the change to
+both the Android and the iOS codebases, or else we'll get runtime errors. With
+three repositories to work with, it was too easy to forget to add one or the
+other (as you're generally developing with just one simulator open), and then
+it breaks the build for other environment.
+
+We'd also get into a state where the `master` branch of `react-native`
+contained some changes that had only been coordinated with e.g. the `ios`
+repo, and the `android` repo's `react-native` submodule would be several
+commits behind. Then someone working on the android side would update the
+submodule, and they'd have to track down all the breakages.
+
+In short, we started running into a lot of synchronization issues, and we
+really wanted to just have things be in the same place, such that pull
+requests could be combined, reviews would be more coherent, and it would be
```

Comment:
Can't overstate how annoying it was to do the three repo dance, with all the dependent changes and Phab revisions and test plans, just for a single atomic change...